src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.health = 500;
+    this.name = 'Test';
+    Animal.alive.push(this.name);
+  }
+  static deleteIndex = Animal.alive.findIndex((animal) => animal.health !== 'dead');
+
+  deleteAnimal() {
+    Animal.alive = Animal.alive.filter((animal) => animal.name = 'dad');
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(name);
+    this.hidden = true;
+  }
+  hide() {
+    return this;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.hidden === false && animal instanceof Animal) {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
+      animal.deleteAnimal();
+    }
```

Comment:
After biting, if the animal's health is less than or equal to 0, it should be removed from the Animal.alive array. The current implementation may not work as intended due to issues with the deleteAnimal method.